src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java

```diff
@@ -0,0 +1,1206 @@
+        }
+
+        public static void ensureScalarFits(byte[] scalar) {
+            if (scalar == null) {
+                throw new IllegalArgumentException(
+                    "Null pointers are not allowed as target arrays");
+            }
+
+            if (scalar.length < RISTRETTO255_SCALAR_BYTES) {
+                throw new IllegalArgumentException(
+                    "To hold a Ristretto255 scalar, the array must be "
+                        + RISTRETTO255_SCALAR_BYTES
+                        + " bytes in size");
+            }
+        }
+
+        public static void checkPoint(byte[] point) {
+            if (point == null) {
+                throw new IllegalArgumentException(
+                    "Null pointers are not allowed as Ristretto255 points");
+            }
+
+            if (point.length != RISTRETTO255_BYTES) {
+                throw new IllegalArgumentException("A Ristretto255 point must be "
+                                                       + RISTRETTO255_BYTES
+                                                       + " bytes in size");
+            }
+        }
+
+        public static void checkHash(byte[] hash) {
+            if (hash == null) {
+                throw new IllegalArgumentException(
+                    "Null pointers are not allowed as Ristretto255 hashes");
+            }
+
+            if (hash.length != RISTRETTO255_HASH_BYTES) {
+                throw new IllegalArgumentException("A hash for use with Ristretto255 must be "
+                                                       + RISTRETTO255_HASH_BYTES
+                                                       + " bytes in size");
+            }
+        }
+
+        public static void checkScalar(byte[] scalar) {
+            if (scalar == null) {
+                throw new IllegalArgumentException(
+                    "Null pointers are not allowed as Ristretto255 scalars");
+            }
+
+            if (scalar.length != RISTRETTO255_SCALAR_BYTES) {
+                throw new IllegalArgumentException("A Ristretto255 scalar must be "
+                                                       + RISTRETTO255_SCALAR_BYTES
+                                                       + " bytes in size");
+            }
+        }
+
+        public static void checkNonReducedScalar(byte[] scalar) {
+            if (scalar == null) {
+                throw new IllegalArgumentException(
+                    "Null pointers are not allowed as non-reduced Ristretto255 scalars");
+            }
+
+            if (scalar.length != RISTRETTO255_NON_REDUCED_SCALAR_BYTES) {
+                throw new IllegalArgumentException("A non-reduced Ristretto255 scalar must be "
+                                                       + RISTRETTO255_NON_REDUCED_SCALAR_BYTES
+                                                       + " bytes in size");
+            }
+        }
+    }
+
+    final class RistrettoPoint {
+
+        private final LazySodium ls;
```

Comment:
Hm a lot of the code relies on hexadecimal strings. Perhaps we can inside `LazySodium.java` have a getter that exposes the `messageEncoder` like so:

```java
public MessageEncoder getMessageEncoder() {
    return messageEncoder;
}
```

and then you can use the message encoder to encode and decode to your liking everywhere 👍

Reply:
* Added a new `encode` method to return the point encoded with the `LazySodium`'s `MessageEncoder`. 
* `toString` now returns the result of `encode`
* The constructor taking a string now uses the `MessageEncoder`
* `toHex` and `fromHex` methods remain as a common use case
* Added a new `fromString` method using the `MessageEncoder`
* Changed some naming
* Added the `hashToPoint` convenience method to provide a single interface for the common task of hashing some input and then calling `cryptoCoreRistretto255FromHash`

Reply:
* Added methods `encodeToString` and `decodeFromString` using the `MessageEncoder` to `LazySodium`